src/components/ZoomableCards/ZoomCardOpened.js

```diff
@@ -0,0 +1,194 @@
+import { RenderFnType } from '../../prop-types'
+
+const LARGE_MARGIN = 30
+const SIDEBAR_WIDTH = 360
+
+class ZoomCardOpened extends React.Component {
+  static propTypes = {
+    cardRect: PropTypes.instanceOf(DOMRect),
+    renderContent: PropTypes.func.isRequired,
+    renderAside: RenderFnType,
+    currentId: PropTypes.string,
+  }
+
+  state = {
+    openedCardRect: null,
+  }
+
+  _openedCard = React.createRef()
+
+  componentDidMount() {
+    this.setState({ openedCardRect: this.getOpenedCardRect() })
+  }
+
+  componentDidUpdate(prevProps) {
+    this.updateCardRect(prevProps.cardRect, this.props.cardRect)
+  }
+
+  updateCardRect(prevCardRect, cardRect) {
+    if (!cardRect || cardRect === prevCardRect) {
+      return
+    }
+
+    this.setState({ cardRect, openedCardRect: this.getOpenedCardRect() })
+  }
+
+  getOpenedCardRect() {
+    return this._openedCard.current.getBoundingClientRect()
+  }
+
+  getTransform = t => {
+    const { cardRect, openedCardRect } = this.state
+
+    const to = openedCardRect
+    if (!to) {
+      return 'none'
+    }
+
+    const from = cardRect
+    if (!from) {
+      return 'none'
+    }
+
+    return `
+      translate3d(
+        ${lerp(t, -to.x + from.x, 0)}px,
+        ${lerp(t, -to.y + from.y, 0)}px,
+        0
+      )
+      scale3d(
+        ${lerp(t, from.width / to.width, 1)},
+        ${lerp(t, from.height / to.height, 1)},
+        1
+      )
+    `
+  }
+  render() {
+    const { renderContent, renderAside, currentId } = this.props
+    return (
+      <ScrollView
+        css={`
+          pointer-events: ${currentId === null ? 'none' : 'auto'};
+        `}
+      >
+        <Viewport>
+          {({ below }) => {
+            const fullscreen = below('medium')
+            const margin = fullscreen ? 0 : LARGE_MARGIN
+            return (
+              <div
+                css={`
+                  display: flex;
+                  padding: ${fullscreen ? 20 : LARGE_MARGIN}px ${margin}px;
+                `}
+              >
+                <div
+                  ref={this._openedCard}
+                  css={`
+                    position: relative;
+                    display: flex;
+                    width: calc(
+                      100% - ${renderAside ? SIDEBAR_WIDTH - margin : 0}px
+                    );
+                    flex-direction: column;
+                  `}
+                >
+                  <Transition
+                    items={currentId}
+                    from={{ openProgress: 0 }}
+                    enter={{ openProgress: 1 }}
+                    leave={{ openProgress: 0 }}
+                    config={springs.smooth}
+                    initial={null}
+                    native
+                  >
+                    {currentId =>
+                      currentId !== null &&
+                      (({ openProgress }) => (
+                        <Card
+                          fullscreen={fullscreen || undefined}
```

Comment:
@bpierre I know we were having issues with passing props to the DOM element before in aragonUI but I'm pretty sure this isn't the right way to fix it :/

Reply:
If you know a better way, I would love to hear about it!

The problem is that if the styled component is wrapping a React component (as it always happens with `animated.div`) styled-components doesn’t know if the component is making use of a prop or not, so it passes everything. But `animated` passes everything too.

There are other solutions (e.g. creating another top level component to filter the props), but I don’t know any “clean” one… until styled-components implements a proper solution to declare this.

In the meantime, another thing I do a lot is to declare what should be in the CSS as inline styles.

https://github.com/styled-components/styled-components/pull/2093
https://github.com/styled-components/styled-components/pull/2392